src/Entities/Floater.cpp

```diff
@@ -191,12 +191,12 @@ void cFloater::Tick(std::chrono::milliseconds a_Dt, cChunk & a_Chunk)
 		{
 			AttachTo(Callback.GetHitEntity());
 			Callback.GetHitEntity()->TakeDamage(*this);  // TODO: the player attacked the mob not the floater.
-			m_AttachedMobID = Callback.GetHitEntity()->GetUniqueID();
+			m_AttachedMobID = static_cast<int>(Callback.GetHitEntity()->GetUniqueID());
```

Comment:
Change the type of `m_AttachedMobID` instead, to match the UniqueID; initialize to `cEntity::INVALID_ID` in constructor.